code/game/machinery/vending.dm

```diff
@@ -281,9 +288,54 @@
 			return;
 		else
 			to_chat(user, "<span class='notice'>You should probably unscrew the service panel first.</span>")
+	else if(item_slot_check(user, I))
+		insert_item(user, I)
+		return
 	else
 		..()
 
+//Override this proc to do per-machine checks on the inserted item, but remember to call the parent to handle these generic checks before your logic!
+/obj/machinery/vending/proc/item_slot_check(mob/user, obj/item/I)

```

Comment:
Add an `istype(I)` check to this

Reply:
I don't see a need for this, as this is a base proc that gets overriden to perform the checks for the item. The base proc (parent to be called) is just checking if the machine is capable of accepting an item in the first place, not caring what it is at the time since that gets handled by the override